alfasim_sdk/types.py

```diff
@@ -248,3 +248,19 @@ def check(  # pylint: disable=arguments-differ
 @attr.s(kw_only=True)
 class Boolean(BaseField):
     value: bool = attrib(validator=instance_of(bool))
+
+
+@attr.s(kw_only=True)
+class FilePath(BaseField):
+    """
+    The FilePath component provides a platform-native file dialog to the user to be able to select a file.
+    The name of the selected file will be available over the GUI e be enabled to be manually changed.
+    Be aware that the provided file from the user could be wrong, or not available when the application
```

Comment:
"wrong"? what do you mean?

Reply:
If the project file is shared the sharing user should be aware that another file should be also shared and the destination user that this other file must be placed, possibly, in another folder?

I don't think this is a good way to tackle this.
Why not load this data into memory then dump that to the project file?
Or at least copy the file to a known an easy to share location?

Reply:
Definitely, thanks for bringing this to attention.

As we discussed, we should copy the file to the data directory of the project, and the plugin accesses that file path. 👍

Reply:
Wow. Could this at least be optional? I mean, if the file is really big (Imagine a 9999GB read-only file), this means that the user is doomed to wait the file copy to always happen?
Also, if the file comes from a read-only server where the user cannot copy stuff from, he may even be unable to use the plugin.

I think assuming the file is available and checking it from the status monitor is actually reasonable, TBH.
Am I thinking too hard here?

Reply:
>  I mean, if the file is really big (Imagine a 9999GB read-only file)

Come on... example by absurd? 😛 

Our usual case is files of reasonable sizes, not Google-size index databases.

> Also, if the file comes from a read-only server where the user cannot copy stuff from, he may even be unable to use the plugin.

And how he would use the file at all, if he can't read it? 

> I think assuming the file is available and checking it from the status monitor is actually reasonable, TBH.

Not really, because if the user transfers the file to another user, the *full path* will be saved, which is most certainly wrong.

> Am I thinking too hard here?

Perhaps. 😁 

We will be talking about files of a few mega bytes in almost all cases. The safe bet is to just copy the file next to the project. It also allows easy sharing of the project file/directory without nasty surprises.

Reply:
Fair enough